### PR TITLE
Fea unified recipe documentation

### PR DESCRIPTION
**Pull request by loliverhennigh**
Added documentation and updated download scripts to make it easier to get up and running with the unified recipe. Following the instructions should allow a user to get training on almost any machine with a GPU having more than 8 GB or memory. 

There is one caveat in this PR. I have configs set up to train on a small chunk of data. This works however I am getting Nans from the AFNO model after a few epochs. I believe this is an issue with the instability in AFNO and the smaller dataset. Following this PR I am getting the SFNO configs in. My plan is to switch the minimal example to SFNO which should be more stable.

### CODE REVIEW

File: examples/weather/unified_recipe/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Unified Recipe for Training Global Weather Forecasting Models
+
+This example demonstrates how to train a neural global weather forecast model.
+The recipe is set up so that modifying the model architecture, data, or the
+training procedure is straightforward.
+
+## Configs
+
+The `conf` directory contains the configuration files for the model, data,
+training, etc... The configs are given in YAML format and use the `omegaconf`
```
Comment (ktangsali) [inline examples/weather/unified_recipe/README.md:10]: Let's just keep one dot. :)

File: examples/weather/unified_recipe/README.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+library to manage them. Several example configs are given for generating
+different datasets, models, and training procedures. For example, AFNO and
+Graphcast are given with corresponding training procedure and datasets configs.
```
Comment (ktangsali) [inline examples/weather/unified_recipe/README.md:13]: nit: I think C is capitalized, i.e. GraphCast

File: examples/weather/unified_recipe/download_era5.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        time=slice(
+            datetime.datetime.strptime(cfg.dataset.years[0], "%Y-%m-%d"),
+            datetime.datetime.strptime(cfg.dataset.years[1], "%Y-%m-%d"),
+        )
+    )
+    arco_era5 = arco_era5.sel(
+        time=arco_era5.time.dt.hour.isin(np.arange(0, 24, cfg.dataset.dt))
+    )
+
+    # save to pbss fs
```
Comment (ktangsali) [inline examples/weather/unified_recipe/download_era5.py:67]: Reference to pbss needs to be removed?

**Review by ktangsali — APPROVED** [review]
This looks great to me! Very lucid explanations. Added a few minor comments. Also, would be good to make an entry to master README about this here: https://github.com/NVIDIA/modulus/tree/main/examples#weather for higher visibility.

File: examples/weather/unified_recipe/curate_era5.py [hunk 4]
```diff
@@ -88,36 +77,36 @@ def __init__(
                 needed_variables.append(variable[0])
             else:
                 needed_variables.append(variable)
-        for variables in self.arco_era5.variables:
+        for variables in self.era5.variables:
```
Comment (Alexey-Kamenev) [inline examples/weather/unified_recipe/curate_era5.py:80]: Why is `variables` plural? Does `self.era5.variables` contain list of lists? If so, the the next line (`not in` check) would not work?

File: examples/weather/unified_recipe/train.py [hunk 5]
```diff
@@ -27,7 +27,9 @@
 from torch.cuda.amp import GradScaler
 from torch.optim.lr_scheduler import SequentialLR
 from torch import nn
+from torch import Tensor
```
Comment (Alexey-Kamenev) [inline examples/weather/unified_recipe/train.py:30]: Same comment as before re: unused/sorted imports.

File: examples/weather/unified_recipe/train.py [hunk 6]
```diff
@@ -103,7 +105,6 @@ def main(cfg: DictConfig) -> None:
             },  # TODO: maybe mobe this conversion to resolver?
```
Comment (Alexey-Kamenev) [inline examples/weather/unified_recipe/train.py:105]: Note: `rank_zero_logger` is not used. Probably should make it consistent with usage of `LaunchLogger`?

**Review by Alexey-Kamenev — APPROVED** [review]
LGTM with some minor comments.

**Comment by loliverhennigh** [comment]
> This looks great to me! Very lucid explanations. Added a few minor comments. Also, would be good to make an entry to master README about this here: https://github.com/NVIDIA/modulus/tree/main/examples#weather for higher visibility.

This is a great idea. I will do this in the SFNO config PR though as that should mark when we are ready to have people use it.

File: examples/weather/unified_recipe/curate_era5.py (hunk 4 above)
Comment (loliverhennigh) [reply examples/weather/unified_recipe/curate_era5.py:80]: Fixed the plural. `arco_era5.variables` is a just list of variable so it works

**Pull request merged**